### DISCUSSION TITLE
fsck

### DISCUSSION THREAD

**Discussion by arikoza** [devmfc/debian-on-amlogic#91] [Q&A]
I have an external drive connected and mounted, sometimes the lights turn off and I need to do fsck -fy /dev/sda1 . How can I make it happen automatically every time I turn it on?

**Comment by NickelNuts** [comment]
If the mount is listed in /etc/fstab, you can check all filesystems at boot by passing the boot parameter in /boot/boot.config.  I believe it would be "bootargs0=fsck.mode=force fsck.repair=yes"  to force a filesystem check and repair before mounting

**Comment by petesimon** [comment]
I am using tag/v6.12.56 debian trixie on a vontar-x4 and a hk1-x4 tv boxes. I don't want to use V6.18.10 because the kernel is too new for a tv box and causes compatibility issues. About every 2 to 3 days, there are data errors on the root / ext4 filesystem. I use the `-n` option to do a read-only scan of the file system. I can experiment with using two possible config lines in /boot/box.config for the s905x4 chip. I am using the factory stock 5 volt, 2 amperes power adapter. I don't have another power adapter.

### So... what can I do to resolve the issue?

`/boot/box.config`
box=vontarx4_gigabit
#_or_
box=s905x4_generic_gigabit

`fsck.ext4 -fn /dev/mmcblk0p2`
e2fsck 1.47.2 (1-Jan-2025)
Warning!  /dev/mmcblk0p2 is mounted.
Warning: skipping journal recovery because doing a read-only filesystem check.
Pass 1: Checking inodes, blocks, and sizes
Pass 2: Checking directory structure
Pass 3: Checking directory connectivity
Pass 4: Checking reference counts
Pass 5: Checking group summary information
Free blocks count wrong for group # 17 (6478, counted=6477).
Fix? no
Free blocks count wrong (14847996, counted=14847586).
Fix? no
Free inodes count wrong (3839109, counted=3839098).
Fix? no
Block bitmap differences: Group 17 block bitmap does not match checksum.
IGNORED.
ROOTFS: ********** WARNING: Filesystem still has errors **********
ROOTFS: 52667/3891776 files (0.1% non-contiguous), 729092/15577088 blocks
